model/sharing/replicator.go

```diff
@@ -578,6 +578,12 @@ func (s *Sharing) sendBulkDocs(inst *instance.Instance, m *Member, creds *Creden
 
 // ApplyBulkDocs is a multi-doctypes version of the POST _bulk_docs endpoint of CouchDB
 func (s *Sharing) ApplyBulkDocs(inst *instance.Instance, payload DocsByDoctype) error {
+	mu := lock.ReadWrite(inst, "sharings/"+s.SID+"/_bulk_docs")
```

Comment:
After thinking a bit, I would prefer to use a different lock for sending and receiving:

```go
mu := lock.ReadWrite(inst, "sharings/"+s.SID+"/apply")
```

For example, if there is a sharing between the instances A and B, and both try to replicate at the same time, they will acquire the lock when the replicate worker starts, will send their requests, but the received HTTP request cannot be processed because the lock cannot be taken (it's already taken by the sender goroutine).

Reply:
Oh, you have made the change while I was doing the review :+1:

Reply:
Actually, the lock has not the same name between sending `sharings/"+s.SID` and receiving `sharings/"+s.SID+"/_bulk_docs`, but I actually changed it few minutes ago by a force-push, so you might have seen the previous commit.
Should I keep the `/_bulk_docs` or do you prefer `apply`?

Reply:
`/_bulk_docs` is OK